penguin/engine.py

```
@@ -1904,268 +1918,60 @@ async def run_task(
-            if message_callback:
-                await message_callback(f"Error executing task: {str(e)}", "error")
+            result = await self._iteration_loop(cm, config, max_iters)
+            result["task"] = task_metadata
 
-            # Publish error event
-            if enable_events:
+            if on_completion:
                 try:
-                    from penguin.utils.events import EventBus, TaskEvent
+                    await on_completion(result)
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)